src/port.js

```diff
@@ -265,7 +265,7 @@ export default class Port {
 	}
 	static validateEvent(targetOrigin, endpoint, e) {
 		var isValid = (e.source === endpoint) &&
-			(targetOrigin === '*' || targetOrigin === e.origin) &&
+			(targetOrigin === '*' || targetOrigin != null && e.origin != null && targetOrigin.toUpperCase() === e.origin.toUpperCase()) &&
```

Comment:
I'm guessing once you add that, your `undefined` cases will start to fail, so you'll need to either explicitly check for `undefined` first in addition to `null`, or:

    ( targetOrigin === '*' || targetOrigin && targetOrigin.toUpperCase() === e.origin.toUpperCase() )

That might simplify things a bit (if it works and passes the tests). It should handle `undefined`, `null` and empty string.